lib/resty/upload.lua

```diff
@@ -174,15 +222,28 @@ local function read_header(self)
 
     local line, err = read_line(self.line_size)
     if err then
+        if self.restore_body_buffer then
+            ngx_finish_body()
+        end
         return nil, nil, err
     end
 
+    if self.restore_body_buffer then
+        ngx_append_body(line .. "\r\n")
+    end
+
     local dummy, err = read_line(1)
     if dummy then
+        if self.restore_body_buffer then
+            ngx_finish_body()
```

Comment:
do we also need to restore that `dummy`?

Reply:
The dummy is there to detect if the line we are trying to read exceeded the size of self.line_size. When this happens the request can't be processed further. This means that the remainder of the request will not be restored anyway. So there is no point is appending the one byte.